src/crypto/crypto_ecdh.cc

```diff
@@ -596,6 +596,58 @@ WebCryptoKeyExportStatus ECKeyExportTraits::DoExport(
   }
 }
 
+// TODO: this needs a new home
+Maybe<bool> ExportJWKOkpKey(
+    Environment* env,
+    std::shared_ptr<KeyObjectData> key,
+    Local<Object> target) {
+  ManagedEVPPKey pkey = key->GetAsymmetricKey();
+  // TODO: CHECK EVP_PKEY_id(pkey.get()) is one of
+  // EVP_PKEY_X448, EVP_PKEY_ED448, EVP_PKEY_X25519, EVP_PKEY_ED25519
+
+  size_t len = 0;
+  EVP_PKEY_get_raw_public_key(pkey.get(), nullptr, &len);
+
+  uint8_t* rawX = new uint8_t[len];
+  EVP_PKEY_get_raw_public_key(pkey.get(), rawX, &len);
+
+  if (target->Set(
+          env->context(),
+          env->jwk_kty_string(),
+          env->jwk_okp_string()).IsNothing()) {
+    return Nothing<bool>();
```

Comment:
This leaks `rawX` – is there any reason not to use a standard container here instead, i.e. `std::vector<uint8_t> raw_x(len);`? ditto below for `rawD` (/`raw_d`)